modules/scripts-utils/dtbook-utils/src/main/java/org/daisy/pipeline/modules/dtbook_utils/impl/DTBookCleanerLibrary.java

```diff
@@ -0,0 +1,32 @@
+package org.daisy.pipeline.modules.dtbook_utils.impl;
+
+import java.util.Locale;
+import java.util.Date;
+import java.text.SimpleDateFormat;
+import org.daisy.common.xpath.saxon.ExtensionFunctionProvider;
+import org.daisy.common.xpath.saxon.ReflexiveExtensionFunctionProvider;
+import org.osgi.service.component.annotations.Component;
+
+
+public class DTBookCleanerLibrary {
+
+    @Component(
+            name = "DTBookCleanerLibrary",
+            service = { ExtensionFunctionProvider.class }
+    )
+    public static class Provider extends ReflexiveExtensionFunctionProvider {
+        public Provider() {
+            super(DTBookCleanerLibrary.class);
+        }
+    }
+
+    public String getDefaultLocale(){
+        return Locale.getDefault().toString().replace('_', '-');
+    }
+
+    public String getDate(){
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+        return sdf.format(new Date());
+    }
```

Comment:
These can be changed to static methods, so that you don't need the `Library:new()` in library.xsl.

Reply:
Done